findIndex.ts

```diff
@@ -2,21 +2,19 @@ import curryN from './utils/curry_n.ts'
 import type { PH } from './utils/types.ts'
 
 // @types
-type FindIndex_2<T> = ((element: T) => number) &
-  ((element?: PH) => FindIndex_2<T>)
+type FindIndex_2<T> = (element: T) => number
 
-type FindIndex_1<T> = ((arr: T[]) => number) &
-  ((arr?: PH) => FindIndex_1<T>)
+type FindIndex_1<T> = (arr: T[]) => number
 
-type FindIndex = (<T>(arr: T[], element: T) => number) &
-  (<T>(arr: T[], element?: PH) => FindIndex_2<T>) &
-  (<T>(arr: PH, element: T) => FindIndex_1<T>) &
-  ((arr?: PH, element?: PH) => FindIndex)
+type FindIndex =
+  & (<T>(arr: T[], element?: PH) => FindIndex_2<T>)
+  & (<T>(arr: PH, element: T) => FindIndex_1<T>)
+  & (<T>(arr: T[], element: T) => number)
 
 // TODO: (singla-shivam) Add transformer
 // TODO: (ch-shubham) Add Support for Predicates
-
-function _findIndex(arr: Array<any>, element: any) {
+//@ts-ignore
```

Comment:
why ts-ignore?
add `<T>` as to the function. See https://www.typescriptlang.org/docs/handbook/generics.html